crates/store/src/state.rs

```diff
@@ -257,7 +257,7 @@ impl State {
             let mut inner = self.inner.write().await;
 
             // Save the block before transaction is committed:
-            self.block_store.save_block(block_num, &block_data)?;
+            self.block_store.save_block(block_num, &block_data).await?;
```

Comment:
Not for this PR, but in the future we could move this call up and to right before we start updating in-memory structure, but await it still here. This way, writing to file could happen at the same time as memory structures are updated (and it is OK if we encounter an error during update, the incorrect file would be overwritten on the next valid update).